apps/full-site-editing/full-site-editing-plugin/blocks/page-content/index.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { find, get, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { registerBlockType } from '@wordpress/blocks';
+import { IconButton, Placeholder, SelectControl, Toolbar } from '@wordpress/components';
+import { compose, withState } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { Fragment, RawHTML } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const edit = compose(
+	withSelect( select => ( {
+		// TODO: Replace with a better way to fetch pages
+		pages: select( 'core' ).getEntityRecords( 'postType', 'page' ),
+	} ) ),
+	withState( {
+		isEditing: false,
+	} )
+)( ( { attributes, isEditing, pages, setAttributes, setState } ) => {
+	const { align, selectedPageId } = attributes;
+
+	const selectOptions = [
+		{ label: '', value: undefined },
+		...map( pages, page => ( {
+			label: page.title.rendered,
+			value: page.id,
+		} ) ),
+	];
+
+	const toggleEditing = () => setState( { isEditing: ! isEditing } );
+
+	const onChange = pageId => {
+		if ( pageId ) {
+			setState( { isEditing: false } );
+		}
+		setAttributes( { selectedPageId: parseInt( pageId, 10 ) } );
+	};
+
+	const selectedPage = find( pages, { id: selectedPageId } );
+
+	const showToggleButton = ! isEditing || !! selectedPageId;
+	const showPlaceholder = isEditing || ! selectedPageId;
+	const showPreview = ! isEditing && !! selectedPageId;
+
+	return (
+		<Fragment>
+			{ showToggleButton && (
+				<BlockControls>
+					<Toolbar>
+						<IconButton
+							className={ classNames( 'components-icon-button components-toolbar__control', {
+								'is-active': isEditing,
+							} ) }
+							label={ __( 'Change Preview' ) }
+							onClick={ toggleEditing }
+							icon="edit"
+						/>
+					</Toolbar>
+				</BlockControls>
+			) }
+			<div
+				className={ classNames( 'wpcom-page-content-block', {
+					[ `align${ align }` ]: align,
+				} ) }
+			>
+				{ showPlaceholder && (
+					<Placeholder
+						icon="layout"
+						label={ __( 'Page Content' ) }
+						instructions={ __( 'Select a page to preview' ) }
+					>
+						<div className="wpcom-page-content-block__selector">
+							<SelectControl
+								onChange={ onChange }
+								options={ selectOptions }
+								value={ selectedPageId }
+							/>
+							{ !! selectedPageId && (
+								<a href={ `?post=${ selectedPageId }&action=edit` }>{ __( 'Edit Page' ) }</a>
+							) }
+						</div>
+					</Placeholder>
+				) }
+				{ showPreview && (
+					<RawHTML className="wpcom-page-content-block__preview">
+						{ get( selectedPage, 'content.rendered' ) }
+					</RawHTML>
+				) }
+			</div>
+		</Fragment>
+	);
+} );
+
+registerBlockType( 'wpcom/page-content', {
+	title: __( 'Page Content Preview' ),
```

Comment:
I'm thinking whether this should be Post Content Preview? 🤔 In the context of `wp_template` CPT, we'd probably want to allow previewing arbitrary post types when viewing the whole template in the editor. So maybe "post" would be more generic term, or just "Content Preview"? (perfectly fine to leave it scoped down for just pages in this PR)

Reply:
I'll keep it like this for now, but I 100% agree that, especially in this context, we should stop mentioning "posts", and just go as generic as possible.

"Content Preview" could sound a bit confusing (content of what?), but "Page (Content) Preview" is unfortunate as well (why just pages?).